libvips/foreign/uhdrload.c

```diff
@@ -102,6 +99,64 @@ typedef struct _VipsForeignLoadUhdrClass {
 G_DEFINE_ABSTRACT_TYPE(VipsForeignLoadUhdr, vips_foreign_load_uhdr,
 	VIPS_TYPE_FOREIGN_LOAD);
 
+/* We need a fast uhdr detector that only looks at the header. We don't want
+ * to pull the whole image in!
+ *
+ * Use vanilla libjpeg only, and just check for an MPF block in an APP2
+ * marker.
+ */
+static int
+vips_foreign_load_uhdr_is_a(VipsSource *source)
+{
+	VipsImage *context = vips_image_new();
+
+	ReadJpeg *jpeg;
+	if (!(jpeg = vips__readjpeg_new(source, context, 1, VIPS_FAIL_ON_NONE,
+			  FALSE, FALSE))) {
+		VIPS_UNREF(context);
+		return 0;
+	}
+
+	/* Here for longjmp() from vips__new_error_exit() during
+	 * cinfo->mem->alloc_small() or jpeg_read_header().
+	 */
+	if (setjmp(jpeg->eman.jmp)) {
+		VIPS_UNREF(context);
+		return 0;
+	}
+
+	if (vips__readjpeg_open_input(jpeg)) {
+		VIPS_UNREF(context);
+		return 0;
+	}
+
+	struct jpeg_decompress_struct *cinfo = &jpeg->cinfo;
+
+	/* Read JPEG header.
+	 */
+	jpeg_save_markers(cinfo, JPEG_APP0 + 2, 0xffff);
```

Comment:
Note to self: investigate use of `jpeg_set_marker_processor()` as a possible follow-up, see:
https://github.com/lovell/sharp/issues/3984#issuecomment-2317449630

Reply:
Good idea!

Reply:
... lets switch to `jpeg_set_marker_processor()` everywhere in a follow-up PR.